src/components/form/field-controls/checkbox-field/types.ts

```diff
@@ -0,0 +1,13 @@
+import { LabelProps } from 'components/form/components'
+
+interface CheckboxFieldProps {
+	errors?: string[]
+	helperText?: string
+	id?: string
+	initialIsChecked?: boolean
+	label: string
```

Comment:
thought: I think ideally we'll want to support an arbitrary React element for `label`, in addition to a string. I know this doesn't align with the spec, but in the case of Tutorial Library specifically we'll also want to be rendering product icons within the label.

Reply:
If you see `CheckboxField` being used there, I can update! From what I saw at a glance, I wasn't sure if it would be used because there won't be things like helper text or error messages.

Reply:
Makes sense! I think we'll want to use it to ensure the base checkbox styles are consistent.

Reply:
Changes:

- Updated `label` to allow `ReactNode` in: https://github.com/hashicorp/dev-portal/pull/872/commits/06faf823f78f10faf5cdf399388090753473c016
- Added an example to the `CheckboxField` Swingset page in: https://github.com/hashicorp/dev-portal/pull/872/commits/c1957d8d1317d48f389bd2301691e2b776686c42